src/main/java/com/health/controller/RestControllerClass.java

```diff
@@ -26,40 +30,43 @@ public class RestControllerClass {
-        this.latestPublishhedTutorial = latestPublishhedTutorial;
-    }
+    @GetMapping("/api/scriptPublished/{tutorialId}/{dateWithTime}")
+    public ResponseEntity<Map<String, String>> isScriptPulished(@PathVariable int tutorialId,
+            @PathVariable Timestamp dateWithTime) {
+        Map<String, String> response = new HashMap<>();
+        Tutorial tutorial = tutService.findByTutorialId(tutorialId);
 
-    @GetMapping("/tutorialsToUploadScripts")
-    public Map<Integer, String> getPublishedTutorial() {
-        Map<Integer, String> resultTutorial = new HashMap<>(latestPublishhedTutorial);
-        Iterator<Map.Entry<Integer, String>> iterator = resultTutorial.entrySet().iterator();
-        while (iterator.hasNext()) {
-            Map.Entry<Integer, String> entry = iterator.next();
-            int tutorialId = entry.getKey();
-            Tutorial tutorial = tutService.findByTutorialId(tutorialId);
-            if (tutorial != null) {
-                ContributorAssignedTutorial conAssignedTutorial = tutorial.getConAssignedTutorial();
-                TopicCategoryMapping topicCat = conAssignedTutorial.getTopicCatId();
-                int catId = topicCat.getCat().getCategoryId();
-                int lanId = conAssignedTutorial.getLan().getLanId();
-                List<Integer> scriptVerList = ServiceUtility.getApiVersion(scriptmanager_api, catId,
-                        tutorial.getTutorialId(), lanId);
+        if (tutorial != null) {
+            ContributorAssignedTutorial conAssignedTutorial = tutorial.getConAssignedTutorial();
+            TopicCategoryMapping topicCat = conAssignedTutorial.getTopicCatId();
+            int catId = topicCat.getCat().getCategoryId();
+            int lanId = conAssignedTutorial.getLan().getLanId();
+            List<Integer> scriptVerList = ServiceUtility.getApiVersion(scriptmanager_api, catId,
+                    tutorial.getTutorialId(), lanId);
 
-                if (scriptVerList != null && scriptVerList.size() > 0) {
+            if (scriptVerList != null && scriptVerList.size() > 0) {
+                response.put("status", "okay");
+                if (tutorial.isStatus()) {
+                    if (tutorial.isAddedQueue()) {
+                        taskProcessingService.addUpdateDeleteTutorial(tutorial, CommonData.UPDATE_DOCUMENT);
+                    } else {
+                        taskProcessingService.addUpdateDeleteTutorial(tutorial, CommonData.ADD_DOCUMENT);
+                    }
 
-                    getLatestPublishhedTutorial().remove(tutorialId);
                 }
+                return ResponseEntity.ok(response);
+            } else {
+                response.put("status", " not published");
+                return ResponseEntity.status(HttpStatus.NOT_FOUND).body(response);
             }
-        }
+        } else {
+            response.put("status", "Tutorial not found");
+            return ResponseEntity.ok(response);
```

Comment:
```suggestion
            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(response);
```
This is not okay. Old code was correct.